mesh/mesh.hpp

```diff
@@ -1259,11 +1259,16 @@ class Mesh
    /** This enumerated type is used to describe interior, boundary, or shared
        interior faces. */
    enum class FaceLocation { Local, Shared, Boundary, NA };
-   /** This enumerated type is used to describe if a face is a conforming face,
-       a non-conforming slave face, or a non-conforming master face. */
+   /** This enumerated type is used to describe if an element face is a
+       conforming or a non-conforming face. In the case of a non-conforming
+       face, the element can either be the coarse element or the fine element of
+       the non-conforming face.
+       By extension, we name a face a non-conforming coarse or fine face,
+       if the face belonging to elem1 is a non-conforming coarse of fine face
+       respectively. */
    enum class FaceConformity { Conforming,
-                               NonConformingMaster,
-                               NonConformingSlave,
+                               NonConformingCoarse,
+                               NonConformingFine,
                                NA
                              };
```

Comment:
We had discussed this separately, but I think it might be a bit confusing to call a face a "fine" face if element 1 is the fine element, and "coarse" if element 1 is the coarse element — the way I think about this, the face is equally fine no matter whether element 1 or element 2 is the fine version, it just depends on the numbering of the elements (it is not an inherent property of the face).

Perhaps clearer would be something like
```
NonConformingFineCoarse
NonConformingCoarseFine
```
which would make the symmetry of the situation clearer.

Reply:
Coarse and fine refers to the elements, not the face, as stated in the documentation, and by the variable names called `elem_*`

Reply:
I was referring to this in the comment:

> By extension, we name a face a non-conforming coarse or fine face, if the face belonging to elem1 is a non-conforming coarse of fine face respectively.

I think this terminology is confusing.

Reply:
This is actually outdated documentation, it is not used any more, I will remove it.